### PR TITLE
Li Threshold: don't offset positive intensities

### PR DESCRIPTION
**Pull request by slavust**
Threshold with the minimal cross-entropy depends on actual thresholded image means. But the means will differ if one offsets image intensities towards zero, which led to significantly inaccurate return value. This can be shown with the provided test.

Since it's still required to handle negative values case, I've ported its handling as it is done in the ITK LiThresholdCalculator.

### CODE REVIEW

File: tests/skimage/filters/test_thresholding.py [hunk 1]
```diff
@@ -377,6 +377,18 @@ def test_li_astro_image():
...
+def test_li_astro_image_no_zeros():
+    # threshold with the minimal cross-entropy depends
+    # non-linearly on means of the thresholded image.
+    image = util.img_as_ubyte(data.astronaut())
+    image[image < 50] = 50
+    threshold = threshold_li(image)
+    ce_actual = _cross_entropy(image, threshold, np.max(image))
+    assert 112 < threshold < 113
+    assert ce_actual < _cross_entropy(image, threshold + 1)
+    assert ce_actual < _cross_entropy(image, threshold - 1)
```
[the remark this answers is not in the document]
Comment (slavust) [reply tests/skimage/filters/test_thresholding.py:389]: fixed

**Review by fallenmi — APPROVED** [review]
I reviewed exact head `f842b08bf935b070e4a885cd5a0cfd1de9620827` and the current merge `10759e795325b30d77b09036bd3da20300e90942` with `main` `91b0179dfe955de6ec8d9b4e93326a26edcf5c7e`.

Keeping already-positive intensities in their original domain makes the Li update use the actual foreground and background means, while the zero-minimum and negative-minimum paths retain the existing shift-to-zero behavior. An independent exact-source oracle exercised 968 positive integer distributions: 456 structured two/three-mode cases and 512 deterministic randomized multimodal cases. On every case, exact head and current merge produced the update implied by the original-intensity means with zero numerical residual; exact base instead applied the shifted-intensity recurrence. The structured cases never increased cross-entropy. On the submitted positive-only astronaut case, base returns `88.2558305`, while head/merge return `112.189164` with lower cross-entropy.

I also checked four zero/negative-minimum dtype controls, 30 scalar-initial-guess boundary cases, and callback values in the caller's intensity domain. Head and merge matched throughout. Both exact-head and current-merge wheels built successfully, and the two mirrored thresholding test files passed 234 tests with two bounded matplotlib skips on each revision. `git diff --check` is clean, and all four live checks are successful.

No blocking findings.

Disclosure: I used OpenAI Codex to inspect the exact revisions, construct and run the recurrence/cross-entropy oracle, build the head and current merge, run the owning tests, and draft this review. I verified the results and conclusion.
